### PR TITLE
initial implementation of generate geodatabase sample

### PR DESCRIPTION
**Pull request by tschie**
sample to generate a geodatabase from a feature service based on the current visible extent

### CODE REVIEW

**Comment by tschie** [comment]
@john0005 Can you please review? Let me know if you don't have scope for this.

File: src/main/java/com/esri/samples/featurelayers/generate_geodatabase/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+<ol>
+  <li>Create a <code>GeodatabaseSyncTask</code> with the URL of the feature service and load it.</li>
+  <li>Create </code>GenerateGeodatabaseParameters</code> specifying things like the extent and whether to include
+  attachments.</li>
+  <li>Create a <code>GenerateGeodatabaseJob</code> with <code>GenerateGeodatabaseJob job = syncTask
+  .generateGeodatabaseAsync(parameters, filePath)</code>. Start the job with <code>job.start()</code>.</li>
+  <li>When the job is done, <code>job.getResult()</code> will return the geodatabase. Inside the geodatabase are
+  feature tables that can be used to add feature layers to the map.</li>
+  </li>Lastly, it is good practice to call <code>syncTask.unregisterGeodatabaseAsync(geodatabase)</code> when
```
Comment (john0005) [inline src/main/java/com/esri/samples/featurelayers/generate_geodatabase/README.md]: First`</li>` should be `<li>`.

File: src/main/java/com/esri/samples/featurelayers/generate_geodatabase/GenerateGeodatabaseSample.java [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+          } catch (InterruptedException | ExecutionException e) {
+            displayMessage("Error generating geodatabase parameters", e.getMessage());
+          } catch (IOException e) {
+            displayMessage("Could not create file for geodatabase", e.getMessage());
+          }
+        });
+      });
+
+      // add the map view and controls to stack pane
+      stackPane.getChildren().addAll(mapView, generateButton, progressBar);
```
Comment (john0005) [inline src/main/java/com/esri/samples/featurelayers/generate_geodatabase/GenerateGeodatabaseSample.java:154]: I would add a label for the progress bar.

File: src/main/java/com/esri/samples/featurelayers/generate_geodatabase/GenerateGeodatabaseSample.java [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+            generateGeodatabaseParameters = parameters.get();
+            generateGeodatabaseParameters.setReturnAttachments(false);
+
+            // temporary file for geodatabase
+            File tempFile = File.createTempFile("gdb" + replica.getAndIncrement(), ".geodatabase");
+            tempFile.deleteOnExit();
+
+            // create and start the job
+            GenerateGeodatabaseJob job = syncTask.generateGeodatabaseAsync(generateGeodatabaseParameters, tempFile
+                .getAbsolutePath());
```
Comment (john0005) [inline src/main/java/com/esri/samples/featurelayers/generate_geodatabase/GenerateGeodatabaseSample.java]: Would place on one line `syncTask.generateGeodatabaseAsync(generateGeodatabaseParameters,tempFile.getAbsolutePath());`

File: src/main/java/com/esri/samples/featurelayers/generate_geodatabase/GenerateGeodatabaseSample.java [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+      SimpleLineSymbol boundarySymbol = new SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, 0xFFFF0000, 5);
+
+      // add a button to generate the geodatabase and a progress bar
+      Button generateButton = new Button("Generate Geodatabase");
+      generateButton.setDisable(true); //wait until sync task loaded
+      ProgressBar progressBar = new ProgressBar();
+      progressBar.setProgress(0.0);
+
+      // create a geodatabase sync task
+      GeodatabaseSyncTask syncTask = new GeodatabaseSyncTask(FEATURE_SERVICE_URL);
```
Comment (john0005) [inline src/main/java/com/esri/samples/featurelayers/generate_geodatabase/GenerateGeodatabaseSample.java]: Can just place the URL right above here.
`String featureServiceURL =
          "https://sampleserver6.arcgisonline.com/arcgis/rest/services/Sync/WildfireSync/FeatureServer";`
I know that in other sample we always put it at the top, not really for sure why this is, but I feel like putting it here is easier for the use to reference.

File: src/main/java/com/esri/samples/featurelayers/generate_geodatabase/GenerateGeodatabaseSample.java [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+        graphicsOverlay.getGraphics().clear();
+
+        // show the extent used as a graphic
+        Envelope extent = mapView.getVisibleArea().getExtent();
+        Graphic boundary = new Graphic(extent, boundarySymbol);
+        graphicsOverlay.getGraphics().add(boundary);
+
+        // create generate geodatabase parameters for the current extent
+        ListenableFuture<GenerateGeodatabaseParameters> parameters = syncTask
+            .createDefaultGenerateGeodatabaseParametersAsync(extent);
```
Comment (john0005) [inline src/main/java/com/esri/samples/featurelayers/generate_geodatabase/GenerateGeodatabaseSample.java]: Would place on single line.
`syncTask.createDefaultGenerateGeodatabaseParametersAsync(extent);`

File: src/main/java/com/esri/samples/featurelayers/generate_geodatabase/GenerateGeodatabaseSample.java [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+
+            // get geodatabase when done
+            job.addJobDoneListener(() -> {
+              if (job.getStatus() == Job.Status.SUCCEEDED) {
+                Geodatabase geodatabase = job.getResult();
+                displayMessage("Geodatabase successfully generated", "Unregistering geodatabase since we're not " +
+                "syncing it here");
+                geodatabase.loadAsync();
+                geodatabase.addDoneLoadingListener(() ->
+                  geodatabase.getGeodatabaseFeatureTables().forEach(ft -> {
```
Comment (john0005) [inline src/main/java/com/esri/samples/featurelayers/generate_geodatabase/GenerateGeodatabaseSample.java]: would check that the geodatabase loaded properly.
`if (geodatabase.getLoadStatus() == LoadStatus.LOADED) {`

**Comment by john0005** [comment]
Some minor changes but looks good.

**Pull request merged**